### PR TITLE
Add support for blockquotes coming from formatters

### PR DESCRIPTION
**Pull request by jgadbois**
This will require formatters to implement a blockquote method.

### CODE REVIEW

File: app/helpers/forem/formatting_helper.rb [hunk 1]
```diff
@@ -8,5 +8,13 @@ module Forem
         simple_format(h(text))
       end
     end
+
+    def as_quoted_text(text)
+      if Forem.formatter
```
Comment (radar) [inline app/helpers/forem/formatting_helper.rb]: How about this instead?

```
if Forem.formatter && Forem.formatter.respond_to?(:blockquote)
```

That would mean that you wouldn't break existing formatters and future formatters wouldn't need to implement a blockquote method.

Comment (jgadbois) [inline app/helpers/forem/formatting_helper.rb]: Yeah, I like that - committed

**Comment by parndt** [comment]
Can we please have tests for this functionality?

Thanks!

**Comment by jgadbois** [comment]
Started on some specs which brought up a problem - all the formatters run the raw markdown through ERB::Util.h which escapes the markdown blockquote formatting (start the line with greater than).

The current commit in the pull request gets around this by wrapping the post in raw() but that's probably not smart and causes an existing spec to fail.

I'm pretty sure the blockquote tag gets escaped using the current functionality too.

Any suggestions?  I was thinking there might be some markdown sanitizer we could use instead of h()?

**Comment by parndt** [comment]
What do you think @radar ?

**Comment by radar** [comment]
Ok, perhaps we should be running the formatter and then running that output through a HTML sanitizer. Any experience with this @parndt?

**Comment by parndt** [comment]
@radar none

**Comment by jgadbois** [comment]
I removed the escaping from the RDiscount gem and added a call to sanitize before outputting the forem post.  Specs (my new ones and existing ones) are passing.

**Comment by jgadbois** [comment]
Sorry some of these pull request commits are on my master in addition to my formatted-quotes branch, so I'll have to have this pull request merged before I can fork off another branch to start working on the email subscriptions.  I'm pretty new to pull requests so sorry for messing that up.

**Comment by jgadbois** [comment]
It should - I'm sanitizing before running through markdown.

**Comment by radar** [comment]
Alright, I'll run through it tonight when I'm home from work.

**Comment by jgadbois** [comment]
@radar - sorry to bug you.  Had a chance to check this out yet?

**Comment by radar** [comment]
No sorry I've not had the time, yesterday's work took longer than I thought and I also spent the morning + evening biking 50km.  

I am in Adelaide this weekend (until the 27th) visiting my family. I will try to find some time to look into it this weekend.

**Comment by radar** [comment]
There's a binary file in this called "core". What is that?

**Comment by radar** [comment]
This seems to not work... when I click "Quote" on a post it gives me the `<blockquote>` wrapping for the text, but when the post is created this tag vanishes.

**Comment by radar** [comment]
I've made [a new commit to forem-rdiscount here](https://github.com/radar/forem-rdiscount/commit/84470d4) which seems to fix this error.

**Comment by jgadbois** [comment]
Thanks for committing this - I'll get started on email notices.

**Pull request merged**